machine1/dbus_test.go

```diff
@@ -17,14 +17,62 @@ limitations under the License.
 package machine1
 
 import (
+	"os/exec"
 	"testing"
 )
 
-// TestNew ensures that New() works without errors.
-func TestNew(t *testing.T) {
-	_, err := New()
+var (
+	machineName = "testMachine"
+)
 
+func mustCreateTestProcess() (pid int) {
+	systemdRun, err := exec.LookPath("systemd-run")
+	if err != nil {
+		panic(err.Error())
+	}
+	sleep, err := exec.LookPath("sleep")
+	if err != nil {
+		panic(err.Error())
+	}
+	cmd := exec.Command(systemdRun, sleep, "5000")
+	err = cmd.Run()
 	if err != nil {
-		t.Fatal(err)
+		panic(err.Error())
+	}
+	return cmd.Process.Pid + 1
```

Comment:
This may race. It may be better to set a unique name on the transient unit and retrieve the `MainPID` from there.